Bottom: 7f706fbcce873a5602c083819fbf2b3775f885cb
Top:    437274657c6a8822a511bae89408c6eddc634e0f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-07 10:16:10 +0000

Attempt some GL optimisation


---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index 4f944f6..383bee5 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -225,7 +225,7 @@ static void draw_cap (double width, int x, int y, double angle, double scale)
 void
 hidgl_draw_line (int cap, double width, int x1, int y1, int x2, int y2, double scale)
 {
-  double angle;
+  double angle = 0.;
   float deltax, deltay, length;
   float wdx, wdy;
   int circular_caps = 0;
@@ -254,10 +254,15 @@ hidgl_draw_line (int cap, double width, int x1, int y1, int x2, int y2, double s
     wdy = deltax * width / 2. / length;
     wdx = -deltay * width / 2. / length;
 
-    if (deltay == 0.)
-      angle = (deltax < 0) ? 270. : 90.;
+    if (hairline)
+      angle = 0;
     else
-      angle = 180. / M_PI * atanl (deltax / deltay);
+      {
+        if (deltay == 0.)
+          angle = (deltax < 0) ? 270. : 90.;
+        else
+          angle = 180. / M_PI * atanl (deltax / deltay);
+      }
 
     if (deltay < 0)
       angle += 180.;
diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 0517d96..92c6fb9 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -1745,6 +1745,7 @@ ghid_port_drawing_area_expose_event_cb (GtkWidget * widget,
   glEnable (GL_BLEND);
   glBlendFunc (GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
 
+  glShadeModel (GL_FLAT);
 //  glEnable(GL_POLYGON_SMOOTH);
 //  glHint(GL_POLYGON_SMOOTH_HINT, [GL_FASTEST, GL_NICEST, or GL_DONT_CARE]);
